sql/colexec: TestRepeatableBatchSource failed

[(sql/colexec).TestRepeatableBatchSource failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1748840&tab=buildLog) on [master@3f9521f565d1d91cafb22ec76dd833532efdf177](https://github.com/cockroachdb/cockroach/commits/3f9521f565d1d91cafb22ec76dd833532efdf177):

Fatal error:
```
panic: runtime error: slice bounds out of range [:10] with capacity 6 [recovered]
	panic: runtime error: slice bounds out of range [:10] with capacity 6 [recovered]
	panic: runtime error: slice bounds out of range [:10] with capacity 6

```

Stack:
```
goroutine 7076 [running]:
testing.tRunner.func1(0xc041a92700)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x29f2fe0, 0xc0408a5e20)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:98 +0x210
panic(0x29f2fe0, 0xc0408a5e20)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/col/coldata.(*memColumn).Copy(0xc041a81ec0, 0x5, 0x2ffb820, 0xc041a81e40, 0x0, 0x0, 0x0, 0x0, 0x0, 0xa, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/col/coldata/vec.eg.go:829 +0x772f
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*RepeatableBatchSource).Next(0xc00031c850, 0x2fad780, 0xc000052120, 0xc00031c850, 0x6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/testutils.go:118 +0x103
github.com/cockroachdb/cockroach/pkg/sql/colexec.TestRepeatableBatchSource(0xc041a92700)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:1047 +0x135
testing.tRunner(0xc041a92700, 0x2bc5298)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestRepeatableBatchSource
--- FAIL: TestRepeatableBatchSource (0.00s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRepeatableBatchSource PKG=./pkg/sql/colexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRepeatableBatchSource.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/colexec/utils_test.go
+++ b/pkg/sql/colexec/utils_test.go
@@ -1042,6 +1042,9 @@ func TestRepeatableBatchSource(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	batch := testAllocator.NewMemBatch([]coltypes.T{coltypes.Int64})
 	batchLen := uint16(10)
+	if coldata.BatchSize() < batchLen {
+		batchLen = coldata.BatchSize()
+	}
 	batch.SetLength(batchLen)
 	input := NewRepeatableBatchSource(testAllocator, batch)
 

